Tests/Unit/MSFT_xVMDvdDrive.tests.ps1

```
@@ -0,0 +1,783 @@
+
+                Mock `
+                    -CommandName Get-VMScsiController `
+                    -Verifiable
+
+                Mock `
+                    -CommandName Get-VMIdeController `
+                    -Verifiable
+
+                # Mocks that should not be called
+                Mock -CommandName Get-VMHardDiskDrive
+
+                It 'should not throw exception' {
+                    $errorRecord = Get-InvalidArguementError `
+                        -ErrorId 'VMControllerDoesNotExistError' `
+                        -ErrorMessage ($LocalizedData.VMControllerDoesNotExistError -f `
+                            $script:VMName,0)
+
+                    { Test-ParameterValid @script:splatAddDvdDriveNoPath } | Should Throw $errorRecord
+                }
+
+                It 'all the get mocks should be called' {
+                    Assert-VerifiableMocks
+                    Assert-MockCalled -CommandName Get-VM -Exactly 1
+                    Assert-MockCalled -CommandName Get-VMScsiController -Exactly 1
+                    Assert-MockCalled -CommandName Get-VMIdeController -Exactly 1
+                }
+            }
+
+            Context 'VM exists, SCSI contrller exists, HD assigned' {
+                # Verifiable (should be called) mocks
+                Mock `
+                    -CommandName Get-VM `
+                    -MockWith { $script:mockGetVM } `
+                    -Verifiable
+
+                Mock `
+                    -CommandName Get-VMScsiController `
+                    -MockWith { $script:mockGetVMScsiController } `
+                    -Verifiable
+
+                Mock `
+                    -CommandName Get-VMHardDiskDrive `
+                    -MockWith { $script:mockGetVMHardDiskDrive } `
+                    -Verifiable
+
+                # Mocks that should not be called
+                Mock -CommandName Get-VMIdeController
+
+                It 'should not throw exception' {
```

Should throw an exception?